### PR TITLE
#G7j46apS-FT(Backend)Assign driver to buses

### PR DESCRIPTION
**Pull request by karigirwa**
### **What does this PR do?**
This PR is for assigning driver to buses

### **Description of Task to be completed?**

- Operator should be able to assign and unassign driver to a bus
   - **PATCH /api/v1/buses/assigndriver/:busId**
   - **PATCH /api/v1/buses/unassigndriver/:busId**

- The driver should receive a bus assignment and unassignment email & in-app notifications.

- The operator should be able to view a paginated list of driver to bus assignments
   - **GET /api/v1/buses/assignedbuses?page={page}&limit={limit}**

### **How should this be manually tested?**

- .You can use [Swagger documentation](https://phantom-caba-ft-assign--ljkwgj.herokuapp.com/api-docs) and follow the instructions below

- By using postman, use this url: https://phantom-caba-ft-assign--ljkwgj.herokuapp.com **(Optional)**

- Login as admin {"email": "**admin@phantom.com**","password":"**admin**"}

- Register new user and assign him/her a role of _driver_

  - Register new user: **POST /api/v1/auth/register**
  - Assign role: **PATCH /api/v1/auth/updateuser**

- Make assignment and unassignment of driver to bus with the endpoint specified above. For each action check the email notification.

**What are the relevant pivotal tracker stories?**
[#G7j46apS](https://trello.com/c/G7j46apS/12-assign-drivers-to-buses)

**Screenshots (if appropriate)**
tst1
tst2
tst3

### CODE REVIEW

**Review by Nkbtemmy — CHANGES_REQUESTED** [review]
You have more than one commits and there few new lines required

**Comment by Nkbtemmy** [comment]
Remember to rebase

**Review by ncutixavier — APPROVED** [review]

**Review by Nkbtemmy — APPROVED** [review]

**Review by Niyiragira — APPROVED** [review]

File: src/controller/busControllers.js [hunk 1]
```diff
@@ -87,6 +92,107 @@ class busController {
...
+                     if (updated) {
+                            const bus= await Model.Bus.findOne({
+                                 where: { id:req.params.id },
+                                 include:[{
+                                     model:Model.User,
+                                     as:'driver',
+                                     attributes:["id","name","email"]
+                                  }]
+                                 });
+                                 console.log(bus)
```
Comment (ceelogre) [inline src/controller/busControllers.js]: remove log messages. Should only be used in development.

File: src/controller/busControllers.js [hunk 2]
```diff
@@ -87,6 +92,107 @@ class busController {
...
+                    const Options ={
+                        userEmail:req.body.email,
+                        subject:'Phantom Unassignment',
+                        message:message.unassignMessage(req.user.name,bus.plate)
+                        
+                      }
+                      emails.sendEmail(Options) 
+                    return res.status(200).json({ 
+                       bus 
+                    });
```
Comment (ceelogre) [inline src/controller/busControllers.js]: This section is similar to the one in `assignDriver` with a few differences. Can you refactor this into a re-usable function?

File: src/controller/userControllers.js [hunk 3]
```diff
@@ -149,9 +149,9 @@ class userController {
         const payload = decode(token);
 
         
-        console.log('payload=',payload)
+        //console.log('payload=',payload)
```
Comment (ceelogre) [inline src/controller/userControllers.js]: No commented-out log lines. Remove them.

File: src/test/testAssign.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            });
+
+
+            it('it should not retrieve assigned buses when limit is undefined', (done) => {
+                chai.request(app)
+                    .get('/api/v1/buses/assignedbuses?page=1&limit=undefined')
+                    .set("Authorization", userMock.token.operator)
+                    .end((err, res) => {
+                        expect(res.statusCode).to.equal(500);
+                        done();
```
Comment (ceelogre) [inline src/test/testAssign.js]: You should not only `expect` for statusCode, you should check for other attributes for example the return `message`

File: src/test/testUtils.js [hunk 5]
```diff
@@ -12,4 +14,9 @@ describe('PHANTOM - TEST UTILS', () => {
         expect(busClassify.classifyBus(mediumVal)).to.be.a('string').eq('medium')
         expect(busClassify.classifyBus(largeVal)).to.be.a('string').eq('large')
     });
+
+    it('it should return signup message', () => {
+        let email='1',password='3'
+        expect(message.signupEmail(email,password)).to.be.a('string')
```
Comment (ceelogre) [inline src/test/testUtils.js:20]: This is a superficial test.

**Review by ceelogre — CHANGES_REQUESTED** [review]
Your work is almost finished, look at how you are setting language settings in the screenshot below cause when I set it to `fr` or `rw` it still returns english. Also. look at the commented out code.
Capture d’écran 2020-11-04 à 08 34 02

**Review by ceelogre — APPROVED** [review]

**Pull request merged**